week3/homework/src/api.js

```diff
@@ -0,0 +1,91 @@
+const uuid = require('uuid');
+const Util = require('./util');
+const FILENAME = 'todos.json';
+const util = new Util(FILENAME);
+
+class Api {
+  async readTodos(req, res) {
+    let todos = await util.read();
+    res.send(todos);
+  }
+
+  async createTodo(req, res) {
+    try {
+      const todos = await util.read();
+      const todo = {
+        description: req.body.todo.description,
+        id: uuid.v4(),
+        done: false,
+      };
+      todos.push(todo);
+      await util.save(todos);
+      res.end('todo was successful created!');
+    } catch (err) {
+      res.end(err);
+    }
+  }
+
+  async updateTodo(req, res) {
+    try {
+      let todos = await util.read();
+      const todo = todos.find(todo => todo.id === req.params.id);
```

Comment:
this line has been repeated in many places it would be nice to have it in it's method as:
```javascript
getTodo(id) {
    return util.todos.findIndex(todo => todo.id === id);
}
```

and where you use this you will pass only the `id` as:
```javascript
const index = this.getTodo(req.params.id)
```

on each method I read I thought the next is going to be a different `.find` method.